### ISSUE TITLE
Feature request: Please add support for JSON schema

### ISSUE THREAD

**Issue by smitreverie** [mac-s-g/react-json-view#115]
It would be a great solution for generating rest API tryout consoles if JSON schema support is added.

**Comment by mac-s-g** [comment]
thanks for the feedback!  I'll take a look at what that feature entails and update here when I have some time.

**Comment by mac-s-g** [comment]
ah, i see. so you'd like to use JSON Schema to add validation to input changes (add/edit to src object).

is that right?

**Comment by smitreverie** [comment]
Yes, also if an item is added to an array the new item should be created based on schema definition and not variable, required fields are not allowed to be deleted, type restriction, useful error generation etc. etc. I think you get my point. react-jsonschema-form does all this but it generates forms.

**Comment by donaldpipowitch** [comment]
It would also be really nice to see descriptions of a property on hover ❤️

**Comment by mac-s-g** [comment]
i'm on pause with this feature for now.

I added [ajv](https://www.npmjs.com/package/ajv) as a dependency which seems to be the best option for json schema validation.  Unfortunately, it almost doubled my build size.  

I can consider including ajv as a peerDependency rather than including it by default.

that means you would have to install it explicitly for the feature to work.

do you think that is an appropriate solution?

**Comment by donaldpipowitch** [comment]
I'd be fine with that. :)

**Comment by smitreverie** [comment]
Or you can add this feature in the form of an extension to the original library.
